test/ExecuteUpdateTest.cpp

```diff
@@ -0,0 +1,259 @@
+  // `LocalVocab`.
+  expectTransformTriplesTemplate(
+      {},
+      {SparqlTripleSimpleWithGraph{
+          Literal("\"foo\""), Iri("<bar>"), Literal("\"foo\""),
+          SparqlTripleSimpleWithGraph::Graph{::Iri("<baz>")}}},
+      testing::ElementsAre(
+          testing::ElementsAre(VocabIndex(0), VocabIndex(1), VocabIndex(0),
+                               LocalVocabIndex(&entryNotInIndex))),
+      testing::SizeIs(1));
+  // A variable in the template (`?f`) is not mapped in the
+  // `VariableToColumnMap`.
+  expectTransformTriplesTemplateFails(
+      {},
+      {SparqlTripleSimpleWithGraph{Literal("\"foo\""), Iri("<bar>"),
+                                   Variable("?f"),
+                                   SparqlTripleSimpleWithGraph::Graph{}}},
+      testing::HasSubstr(
+          "Assertion `variableColumns.contains(component.getVariable())` "
+          "failed."));
+  expectTransformTriplesTemplateFails(
+      {},
+      {SparqlTripleSimpleWithGraph{
+          Literal("\"foo\""), Iri("<bar>"), Literal("\"foo\""),
+          SparqlTripleSimpleWithGraph::Graph{Variable("?f")}}},
+      testing::HasSubstr("Assertion `variableColumns.contains(var)` failed."));
+  // Variables in the template are mapped to their column index.
+  expectTransformTriplesTemplate(
+      {{Variable("?f"), {0, ColumnIndexAndTypeInfo::PossiblyUndefined}}},
+      {SparqlTripleSimpleWithGraph{Literal("\"foo\""), Iri("<bar>"),
+                                   Variable("?f"),
+                                   SparqlTripleSimpleWithGraph::Graph{}}},
+      testing::ElementsAre(testing::ElementsAre(VocabIndex(0), VocabIndex(1),
+                                                0UL, VocabIndex(2))),
+      testing::IsEmpty());
+  expectTransformTriplesTemplate(
+      {{Variable("?f"), {0, ColumnIndexAndTypeInfo::PossiblyUndefined}}},
+      {SparqlTripleSimpleWithGraph{
+          Literal("\"foo\""), Iri("<bar>"), Literal("\"foo\""),
+          SparqlTripleSimpleWithGraph::Graph{Variable("?f")}}},
+      testing::ElementsAre(testing::ElementsAre(VocabIndex(0), VocabIndex(1),
+                                                VocabIndex(0), 0UL)),
+      testing::IsEmpty());
+}
+
+TEST(ExecuteUpdate, resolveVariable) {
+  auto VocabIndex = [](const uint64_t index) {
+    return Id::makeFromVocabIndex(VocabIndex::make(index));
+  };
+  const auto idTable = makeIdTableFromVector(
+      {{VocabIndex(0), VocabIndex(1), VocabIndex(2)},
+       {VocabIndex(3), VocabIndex(4), VocabIndex(5)},
+       {VocabIndex(6), Id::makeUndefined(), VocabIndex(8)}});
+  EXPECT_THAT(ExecuteUpdate::resolveVariable(idTable, 0, VocabIndex(10)),
+              testing::Eq(VocabIndex(10)));
+  EXPECT_THAT(ExecuteUpdate::resolveVariable(idTable, 0, 1UL),
+              testing::Eq(VocabIndex(1)));
+  EXPECT_THAT(ExecuteUpdate::resolveVariable(idTable, 1, 1UL),
+              testing::Eq(VocabIndex(4)));
+  EXPECT_THAT(ExecuteUpdate::resolveVariable(idTable, 2, 1UL),
```

Comment:
There is a lot of verbose duplication here.
1. (globally in the public namespace) add an `auto V = VocabIndex` and use it
2. (here) bind the `idTable`to the `resolve` call via a lambda etc, to reduce the duplication.